Entity/Components/EnergyComponent.gd

```diff
@@ -0,0 +1,43 @@
+extends EntityComponent
+class_name EnergyComponent
+
+@export var Max_energy: int = 4
+@export var starting_energy: int = 3
+@export var energy_generation: int = 3
+@export var ignore_cost: bool = false
+var energy = starting_energy
+
+signal on_energy_changed
+
+func is_playable(card: CardWorld) -> bool:
+	if ignore_cost:
+		return true
+
+	if card != null:
+		return energy >= card.card_data.energy.energy_cost
+
+	return true
+
+
+func use_energy(card: CardBase) -> void:
+	energy -= card.energy.energy_cost
+
+	if energy < 0 and !ignore_cost:
+		print_stack()
+		push_error("energy cannot be negative")
+
+	on_energy_changed.emit()
+
+
+func on_turn_end() -> void:
+
+	energy += energy_generation
+
+	if energy >= Max_energy:
+		energy = Max_energy
+
+	on_energy_changed.emit()
+
+func add_energy(value: int) -> void:
+	energy += value
```

Comment:
So if you add energy from a card then it goes above the `max_energy`? Is that by design?

Reply:
The energy only returns to the maximum when the turn ends (or starts. after I change it)

Reply:
It's indeed by design. You can check the explanation on the related issue #56